client/app/components/users/UserEdit.jsx

```diff
@@ -0,0 +1,287 @@
+    });
+  };
+
+  regenerateApiKey = () => {
+    const doRegenerate = () => {
+      this.setState({ regeneratingApiKey: true });
+      User.regenerateApiKey(this.state.user).then((apiKey) => {
+        if (apiKey) {
+          const { user } = this.state;
+          this.setState({ user: { ...user, apiKey } });
+        }
+      }).finally(() => {
+        this.setState({ regeneratingApiKey: false });
+      });
+    };
+
+    Modal.confirm({
+      title: 'Regenerate API Key',
+      content: 'Are you sure you want to regenerate?',
+      okText: 'Regenerate',
+      onOk: doRegenerate,
+      maskClosable: true,
+      autoFocusButton: null,
+    });
+  };
+
+  toggleUser = () => {
+    const { user } = this.state;
+    const toggleUser = user.isDisabled ? User.enableUser : User.disableUser;
+
+    this.setState({ togglingUser: true });
+    toggleUser(user).then((data) => {
+      if (data) {
+        this.setState({ user: User.convertUserInfo(data.data) });
+      }
+    }).finally(() => {
+      this.setState({ togglingUser: false });
+    });
+  };
+
+  saveUser = (values, successCallback, errorCallback) => {
+    const data = {
+      id: this.props.user.id,
+      ...values,
+    };
+
+    User.save(data, (user) => {
+      successCallback('Saved.');
+      this.setState({ user: User.convertUserInfo(user) });
+    }, (error = {}) => {
+      errorCallback(error.data && error.data.message || 'Failed saving.');
+    });
+  };
+
+  updatePassword = (values, successCallback, errorCallback) => {
+    if (values.password === values.password_repeat) {
+      this.saveUser(values, successCallback, errorCallback);
+    } else {
+      errorCallback('Passwords don\'t match.');
+    }
+  }
+
+  renderBasicInfoForm() {
+    const { user } = this.state;
+    const formFields = [
+      {
+        name: 'name',
+        title: 'Name',
+        type: 'text',
+        initialValue: user.name,
+        required: true,
+      },
+      {
+        name: 'email',
+        title: 'Email',
+        type: 'email',
+        initialValue: user.email,
+        required: true,
+      },
+    ];
+
+    return (
+      <DynamicForm
+        fields={formFields}
+        readOnly={user.isDisabled}
+        onSubmit={this.saveUser}
+      />
+    );
+  }
+
+  renderChangePassword() {
+    const fields = [
+      { name: 'old_password', title: 'Current Password' },
+      { name: 'password', title: 'New Password', minLength: 6 },
+      { name: 'password_repeat', title: 'Repeat New Password' },
+    ].map(field => ({ ...field, type: 'password', required: true }));
+
+    return (
+      <Fragment>
+        <Modal
+          visible={this.state.passwordModalIsOpen}
+          title="Change Password"
+          onCancel={() => { this.setState({ passwordModalIsOpen: false }); }}
+          footer={null}
+          destroyOnClose
+        >
+          <DynamicForm fields={fields} saveText="Update Password" onSubmit={this.updatePassword} />
```

Comment:
@gabrieldutra can you make the form submit button be the modal ok button?

Reply:
(reason: the button looks inconsistent with the rest of the modals)

Reply:
Makes sense, I could by using `wrappedComponentRef` with `onOk` pointing to `handleSubmit` (sth similar to [this](https://ant.design/components/form/#components-form-demo-form-in-modal)). Also, we will need a workaround to trigger the enter key submit.

Reply:
I wonder if our [latest dialog opening convention](https://github.com/getredash/redash/commit/23f5dde4887eb6a9baa58214e3ac2c0d787b2e98) could be utilized here. This is how it operates currently:
```jsx
import ChangePassDialog from './ChangePassDialog';

openModal() {
  const { result, close, dismiss } = ChangePassDialog.showModal(optionalData);

  // Promise with onOK, onCancel callbacks
  result.then((result) => {
      // modal confirmed with optional result
    })
    .catch(() => {
      // modal dismissed
    });

  close(result); // force dialog confirm (with optional result)
  dismiss(); // force dialog dismiss
}

<button onClick={openModal}></button>
```

Reply:
Interesting... I missed that PR :sweat_smile:. Anyway, I'll try to apply this convention.

Reply:
@ranbena I created a `ChangePasswordDialog` and moved the related logic.

I didn't use `DynamicForm` to be able to make it work with Ant modal buttons and have the "Password don't match" in the field. Unfortunately it's now way more complex, so if you have any ideas to simplify it, LMK :slightly_smiling_face:

![change-password](https://user-images.githubusercontent.com/3356951/52732531-99546680-2fa7-11e9-9e5f-5161b42ea5c1.gif)

Reply:
Wise choice. Looks perfect.